### DISCUSSION TITLE
Sphere trace for Foot IK

### DISCUSSION THREAD

**Discussion by emrehan98** [Sixze/ALS-Refactored#18] [Ideas]
Hi, Sixze. 

I'm not sure if my suggestion is out of scope but there is a problem with line traces. I noticed this problem after I switched to 4.26. Sometimes line traces miss certain type of meshes, in my case the structure mesh. So if I use line trace for IK, it doesn't hit the mesh time to time and it ends up like in the video. So I switched to sphere trace instead of line trace. Maybe giving another trace type as an option can be really helpful in these scenerios.

I guess sphere trace can hit wider range of area so it doesn't miss the target like line trace does.

https://user-images.githubusercontent.com/19621279/133499129-44cc992e-7903-4e8e-b68d-eb62532fe2d1.mp4

Edit: By the way I'm not sure if you are interested with my suggestion but if you are, here's the same static mesh you can test.

[S_Twig_Base_Foundation.zip](https://github.com/Sixze/ALS-Refactored/files/7172929/S_Twig_Base_Foundation.zip)

**Comment by Sixze** [comment]
Foot uses complex traces, and your geometry has a lot of small gaps, so the trace can sometimes go through them. You can try to work around this problem by using a simple collision as complex for this mesh.

![image](https://user-images.githubusercontent.com/16257871/133648566-519cbe6a-1515-4829-896c-c12f3e87bed2.png)

**Reply by Sixze** [reply]
Later I will try to use sphere traces instead, but this could potentially lead to new problems.

**Reply by emrehan98** [reply]
I had no idea about this, switched back to line trace as your solution works perfectly! Thanks.
